python/iceberg/core/filesystem/local_filesystem.py

```diff
@@ -51,34 +63,19 @@ def fix_path(path):
             path = str(path[7:])
         return path
 
-    # def ls(self, path):
-    #     return os.listdir(path)
-    #
-    # def delete(self, path, recursive=False):
-    #     return os.remove(path)
-    #
-
-    #
-    # def rename(self, path, new_path):
-    #     return os.rename(path, new_path)
-    #
-    # def mkdir(self, path, create_parents=True):
-    #     if create_parents:
-    #         return os.makedirs(path)
-    #
-    #     return os.mkdir(path)
-    #
-    # def exists(self, path):
-    #     return os.path.isfile(path)
-    #
-    # def isdir(self, path):
-    #     return os.path.isdir(path)
-    #
-    # def isfile(self, path):
-    #     return os.path.isfile(path)
-    #
-    # def _isfilestore(self):
-    #     return True
-    #
-    # def open(self, path, mode='rb'):
-    #     return open(path, mode=mode)
+    def create(self, path, overwrite=False):
+        if os.path.exists(path) and not overwrite:
+            raise RuntimeError("Path %s already exists" % path)
+
+        return open(path, "w")
+
+    def rename(self, src, dest):
+        try:
+            os.rename(src, dest)
+        except OSError:
+            return False
+
+        return True
+
+    def exists(self, path):
+        return os.path.isfile(path)
```

Comment:
I think you want `os.path.exists` here otherwise we're not particularly clear about the semantics of what we are checking for existence.

Reply:
changed this.